infra/embedding_step_functions/unified_embedding/handlers/mark_batches_complete.py

```
@@ -0,0 +1,206 @@
+                "Batch methods not available, falling back to individual calls",
+                error=str(e),
+            )
+            for batch_id in batch_ids:
+                try:
+                    # Try get_batch_summary first
+                    batch_summary = dynamo_client.get_batch_summary(batch_id)
+                    batch_summary.status = "COMPLETED"
+                    dynamo_client.update_batch_summary(batch_summary)
+                    marked_count += 1
+                    logger.debug("Marked batch as complete", batch_id=batch_id)
+                except Exception as e2:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)